src/Operation/Watch.php

```diff
@@ -130,63 +122,51 @@ public function __construct(Manager $manager, $databaseName, $collectionName, ar
      * @see Executable::execute()
      * @param Server $server
      * @return ChangeStream
-     * @throws UnexpectedValueException if the command response was malformed
-     * @throws UnsupportedException if collation, read concern, or write concern is used and unsupported
+     * @throws UnsupportedException if collation or read concern is used and unsupported
      * @throws DriverRuntimeException for other driver errors (e.g. connection errors)
      */
     public function execute(Server $server)
     {
-        $command = $this->createCommand();
-
-        $cursor = $command->execute($server);
-
-        return new ChangeStream($cursor, $this->createResumeCallable());
-    }
-
-    private function createAggregateOptions()
-    {
-        $aggOptions = array_intersect_key($this->options, ['batchSize' => 1, 'collation' => 1, 'maxAwaitTimeMS' => 1]);
-        if ( ! $aggOptions) {
-            return [];
-        }
-        return $aggOptions;
-    }
+        $cursor = $this->aggregate->execute($server);
 
-    private function createChangeStreamOptions()
-    {
-        $csOptions = array_intersect_key($this->options, ['fullDocument' => 1, 'resumeAfter' => 1]);
-        if ( ! $csOptions) {
-            return [];
-        }
-        return $csOptions;
+        return new ChangeStream($cursor, $this->resumeCallable);
     }
 
     /**
-     * Create the aggregate pipeline with the changeStream command.
+     * Create the aggregate command for creating a change stream.
+     *
+     * This method is also used to recreate the aggregate command if a new
+     * resume token is provided while resuming.
      *
-     * @return Command
+     * @return Aggregate
      */
-    private function createCommand()
+    private function createAggregate()
     {
-        $changeStreamArray = ['$changeStream' => $this->createChangeStreamOptions()];
-        array_unshift($this->pipeline, $changeStreamArray);
+        $changeStreamOptions = array_intersect_key($this->options, ['fullDocument' => 1, 'resumeAfter' => 1]);
+        $changeStream = ['$changeStream' => (object) $changeStreamOptions];
 
-        $cmd = new Aggregate($this->databaseName, $this->collectionName, $this->pipeline, $this->createAggregateOptions());
+        $pipeline = $this->pipeline;
+        array_unshift($pipeline, $changeStream);
 
-        return $cmd;
+        $aggregateOptions = array_intersect_key($this->options, ['batchSize' => 1, 'collation' => 1, 'maxAwaitTimeMS' => 1, 'readConcern' => 1, 'readPreference' => 1, 'session' => 1]);
+
+        return new Aggregate($this->databaseName, $this->collectionName, $pipeline, $aggregateOptions);
     }
 
-    private function createResumeCallable()
+    private function createResumeCallable(Manager $manager)
```

Comment:
After introducing `createAggregate()` so that Aggregate could be constructed in `Watch::__construct()`, this method no longer changed state (with `array_shift($this->pipeline)` removed). That meant we could create the callable once and re-use it from `execute()`.

Saving the callable as a class property then replaces the Manager property, since we can `use` it just for this closure that requires it.